src/mame/drivers/magicard.cpp

```diff
@@ -350,313 +571,656 @@ TODO: check this register,doesn't seem to be 100% correct.
 */
 
 
-void magicard_state::video_start()
+uint32_t magicard_base_state::screen_update_magicard(screen_device &screen, bitmap_rgb32 &bitmap, const rectangle &cliprect)
 {
-}
+	uint8_t *src;
+	uint32_t *dest;
+	int scanline = screen.vpos();
```

Comment:
Not sure if this is the right drawing approach, but anyway: `VIDEO_UPDATE_SCANLINE` will automatically call this multiple times via the cliprect struct. If `scanline < 32` is for avoiding drawing outside the bitmap range (which causes a crash), reading from `cliprect.min_y` / `cliprect.max_y` will ensure that and avoids the extra step here.

Reply:
Yeah, it’s a bad idea to be relying on `vpos` here – just draw the lines specified in the clipping rectangle, and set the necessary flags on the screen if you need your update function to always be called.